sda-auth/info.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/kataras/iris/v12"
+	"github.com/neicnordic/crypt4gh/keys"
+	log "github.com/sirupsen/logrus"
+)
+
+type Info struct {
+	ClientID  string `json:"client_id"`
+	OidcURI   string `json:"oidc_uri"`
+	PublicKey string `json:"public_key"`
+	InboxURI  string `json:"inbox_uri"`
+}
+
+func readPublicKeyFile(filename string) (key *[32]byte, err error) {
+	log.Info("Reading Public key file")
+	file, err := os.Open(filepath.Clean(filename))
+	if err != nil {
+		return nil, err
+	}
+
+	publicKey, err := readPublicKey(file)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading public key file %s: %v", filename, err)
+	}
+
+	return &publicKey, err
+}
+
+// Wrapper for the respective crypt4gh function. Since this function panics if the key is
+// malformed, so we handle that as well as errors.
+func readPublicKey(reader io.Reader) (key [32]byte, err error) {
+
+	defer func() {
+		if recover() != nil {
+			err = fmt.Errorf("malformed key file")
+		}
+	}()
+
+	publicKey, err := keys.ReadPublicKey(reader)
+
+	return publicKey, err
+}
```

Comment:
If we merge and you update to code including https://github.com/neicnordic/crypt4gh/pull/85, this wart shouldn't be needed, methinks.